src/main.rs

```diff
@@ -1,22 +1,59 @@
+extern crate isatty;
+extern crate rustc_demangle;
+extern crate tempdir;
+#[macro_use]
+extern crate structopt;
+
+use isatty::stderr_isatty;
+use rustc_demangle::demangle;
 use std::collections::HashMap as Map;
 use std::env;
 use std::ffi::{OsStr, OsString};
 use std::fs::{self, File};
-use std::io::{self, Read, Write, ErrorKind};
+use std::io::{self, ErrorKind, Read, Write};
 use std::path::{Path, PathBuf};
-use std::process::{self, Child, Stdio, Command};
-
-extern crate isatty;
-use isatty::stderr_isatty;
-
-extern crate rustc_demangle;
-use rustc_demangle::demangle;
-
-extern crate tempdir;
+use std::process::{self, Child, Command, Stdio};
+use structopt::StructOpt;
 use tempdir::TempDir;
 
+#[derive(StructOpt, Debug)]
+#[structopt(
+    name = "cargo-llvm-lines",
+    bin_name = "cargo",
+    about = "Print amount of lines of LLVM IR that is generated for the current project"
+)]
+enum Opt {
+    #[structopt(
+        name = "llvm-lines", raw(setting = "structopt::clap::AppSettings::AllowExternalSubcommands")
+    )]
+    LLVMLines {
+        #[structopt(long = "filter-cargo", raw(hidden = "true"))]
+        filter_cargo: bool,
+        #[structopt(long = "lib", raw(hidden = "true"))]
+        lib: bool,
+
+        #[structopt(long = "bin", raw(hidden = "true"))]
+        bin: Option<String>,
+
+        /// Set the sort order to number of instantiations
+        #[structopt(short = "i", long = "sort-insts")]
```

Comment:
Please pick a different name for this flag. "Inst" commonly stands for "instruction" not "instantiation", especially in the context of LLVM. For example [`InstCombine`](http://llvm.org/doxygen/classllvm_1_1InstCombinePass.html), [`MCInst`](http://llvm.org/doxygen/classllvm_1_1MCInst.html), [`RISCVInstPrinter`](http://llvm.org/doxygen/classllvm_1_1RISCVInstPrinter.html).

Reply:
Sure, any suggestions? I have very little opinion on this

Reply:
[Bloaty](https://github.com/google/bloaty) has a `-s <sortby>` flag which seems reasonable to me. In our case it could be `-s lines` (the default) or `-s copies` with long equivalents `--sort=lines` and `--sort=copies`.

Reply:
Changed it to that